src/main/java/eu/xenit/alfred/content/gateway/GatewayApplication.java

```diff
@@ -146,8 +157,16 @@ public OpaClient opaClient(OpaProperties opaProperties) {
 
     @Bean
     @ConditionalOnBean(OpaClient.class)
-    public PolicyDecisionPointClient pdpClient(OpaProperties opaProperties, OpaClient opaClient) {
-        return new OpenPolicyAgentPDPClient(opaClient, request -> opaProperties.getQuery());
+    public PolicyDecisionPointClient pdpClient(
+            OpaProperties opaProperties,
+            ServiceDiscoveryProperties serviceDiscoveryProperties,
+            OpaClient opaClient,
+            ServiceTracker serviceTracker) {
+        if (serviceDiscoveryProperties.isEnabled()) {
+            return new OpenPolicyAgentPDPClient(opaClient, request -> serviceTracker.opaQueryFor(request));
+        } else {
+            return new OpenPolicyAgentPDPClient(opaClient, request -> opaProperties.getQuery());
```

Comment:
the bigger idea is that you should create a bean of type `OpaQueryProvider` (conditionally) and use that as a dependency, not this if/else logic here